### DISCUSSION TITLE
Cross-origin blocking is much too strict (empty responses at localhost:xxxx)

### DISCUSSION THREAD

**Discussion by actuallymentor** [cypress-io/cypress#28028] [General]
Hello all,

I'm currently locked into cypress v`11.2.0` because once I upgrade to `13.x` Cypress stops being able to call `localhost:8081` 
and fails with `net::ERR_EMPTY_RESPONSE` (firebase emulator) and breaks with a `chrome-error://chromewebdata/` when trying to open a `localhost:5051` (firebase functions emulator).

I've has `chromeWebSecurity` disabled for a long time now, but for some reason, upgrading cypress always gives me more headaches.

I do not have the time to make a reproducible issue but want to post this in case there are others with this issue.

**Comment by AndriiNeverov** [comment]
I face a similar issue too. A call to emulated Firebase Auth fails within the Cypress, but works perfectly fine against the very same Cypress started app from regular Chrome.

`chromeWebSecurity: false` and restarting doesn't make a difference.

I noticed some things different in the Cypress case:

1) "Origin: http://localhost:4200" header is missing
2) "Proxy-Connection: keep-alive" instead of "Connection: keep-alive"

![Screenshot 2024-07-23 at 10 11 19 PM](https://github.com/user-attachments/assets/09f8651e-84b2-4584-ae4e-8662de3cee71)
![Screenshot 2024-07-23 at 10 10 26 PM](https://github.com/user-attachments/assets/b91156aa-8c07-49b7-a6a8-fc310db6307b)

**Comment by sj-net** [comment]
I have spent 2 days to find the issue. I am using Cypress, Firebase Emulator, react and Vite. I was using localhost for emulator connections. And that was the mistake. All emulator requests with localhost are giving `net::ERR_EMPTY_RESPONSE`. And the solution is to use `127.0.0.1` and everything worked as it should. Looks like Cypress is handling the localhost URL's differently. I think they only handle the ports they know, and ignore the other ports.

**Reply by AndriiNeverov** [reply]
This is brilliant!

`127.0.0.1` works like a charm.

**Reply by pablopastorino** [reply]
I have spent 2 days to find your solution.

**Reply by ericbf** [reply]
Worked for me too. Definitely sounds like a defect.
